### PR TITLE
[TD] Query Github API for base

### PR DESCRIPTION
**Pull request by clee2000**
A better query for the base commit of a PR.  
Some ghstack PRs are not connected to main so git merge-base doesn't work.  Instead, use the Github API to query for the base of the PR, which should be more accurate

Sanity checked on one of Ed's ghstack PRs

### CODE REVIEW

File: tools/testing/target_determination/heuristics/utils.py [hunk 1]
```diff
@@ -20,20 +22,44 @@ def python_test_file_to_test_name(tests: Set[str]) -> Set[str]:
...
+@lru_cache(maxsize=None)
+def get_merge_base() -> str:
+    pr_number = get_pr_number()
+    if pr_number is not None:
+        github_token = os.environ.get("GITHUB_TOKEN")
+        headers = {
+            "Accept": "application/vnd.github.v3+json",
+            "Authorization": f"token {github_token}",
+        }
+        url = f"https://api.github.com/repos/pytorch/pytorch/pulls/{pr_number}"
```
Comment (seemethere) [inline tools/testing/target_determination/heuristics/utils.py:46]: Are we worried about rate limiting here?

Comment (clee2000) [reply tools/testing/target_determination/heuristics/utils.py:46]: From what I can find on rockset, we usually run ~30 pull workflows per hour, max I've seen so far is 50.  This only runs once per workflow, so probably a max of 200 per hour (I've only seen 120 max on rockset).  iirc rate limits are 5000 per person per hour so I think it'll be fine

**Review by seemethere — APPROVED** [review]

**Pull request closed without merging**